libqtile/core/manager.py

```diff
@@ -327,23 +327,21 @@ def paint_screen(self, screen, image_path, mode=None):
     def process_key_event(self, keysym: int, mask: int) -> None:
         key = self.keys_map.get((keysym, mask), None)
         if key is None:
-            logger.info("Ignoring unknown keysym: {keysym}, mask: {mask}".format(keysym=keysym, mask=mask))
+            logger.warning("Ignoring unknown keysym: {keysym}, mask: {mask}".format(keysym=keysym, mask=mask))
             return
 
         if isinstance(key, KeyChord):
             self.grab_chord(key)
         else:
+            if self.current_chord and (self.current_chord.mode == "" or key.key == "Escape"):
+                self.ungrab_chord()
```

Comment:
Question: What is the benefit of this being above the key handling? I think this would trigger `hook.fire("leave_chord")` before the actual key actions are triggered?